python/cudf/cudf/core/column/datetime.py

```diff
@@ -82,10 +83,19 @@ def from_numpy(cls, array):
                 ("Cannot infer datetime dtype " + "from np.array dtype `%s`")
                 % (array.dtype)
             )
+        null = cudf.Series(np.array([None]))
+        col = libcudf.replace.replace(
+            as_column(Buffer(array)),
+            as_column(
+                Buffer(np.array([np.datetime64("NaT")], dtype=array.dtype))
+            ),
+            null._column.astype(array.dtype),
+        )
         if cast_dtype:
             array = array.astype(np.dtype("datetime64[ms]"))
         assert array.dtype.itemsize == 8
-        return cls(data=Buffer(array), dtype=array.dtype)
+
+        return cls(data=Buffer(array), mask=col.mask, dtype=array.dtype)
```

Comment:
This logic still needs to be put into https://github.com/rapidsai/cudf/blob/branch-0.10/python/cudf/cudf/core/column/column.py#L1186-L1194 to handle where you're passed a datetime64 gpu array.

Reply:
@kkraus14 `find_and_replace_all` doesn't support NaN as of now.